### PR TITLE
explicitly dump entire XML fragments to help with debugging

### PR DESCRIPTION
**Pull request by brettfo**
This is to potentially help diagnose #9653, where it's expected that the resultant XML fragment is 'correct', but out of order.

### CODE REVIEW

File: src/VisualStudio/Core/Test/Progression/ProgressionTestHelpers.vb [hunk 1]
```diff
@@ -36,10 +36,19 @@ Namespace Microsoft.VisualStudio.LanguageServices.UnitTests.Progression
 
         Public Sub AssertSimplifiedGraphIs(graph As Graph, xml As XElement)
             Dim graphXml = graph.ToSimplifiedXDocument()
-            If Not XNode.DeepEquals(graphXml, xml) Then
+            If Not XNode.DeepEquals(graphXml.Root, xml) Then
```
Comment (brettfo) [inline src/VisualStudio/Core/Test/Progression/ProgressionTestHelpers.vb]: This check was incorrect and always failing, forcing a string comparison every time.

**Comment by brettfo** [comment]
@dotnet/roslyn-ide
@dotnet/roslyn-infrastructure

File: src/VisualStudio/Core/Test/Progression/ProgressionTestHelpers.vb [hunk 2]
```diff
@@ -36,10 +36,19 @@ Namespace Microsoft.VisualStudio.LanguageServices.UnitTests.Progression
...
                 ' They aren't equal, so therefore the text representations definitely aren't equal.
                 ' We'll Assert.Equal those, so that way xunit will show nice before/after text
-                Assert.Equal(xml.ToString(), graphXml.ToString())
+                'Assert.Equal(xml.ToString(), graphXml.ToString())
+
+                ' In an attempt to diagnose some flaky tests, the whole contents of both objects will be output
+                Throw New Exception($"Graph XML was not equal, check for out-of-order elements.
```
Comment (jasonmalinowski) [inline src/VisualStudio/Core/Test/Progression/ProgressionTestHelpers.vb]: Why was this necessary over the default xunit output?

Comment (brettfo) [inline src/VisualStudio/Core/Test/Progression/ProgressionTestHelpers.vb]: Because the current output (taken from the bug) doesn't contain enough info, it's just:

```
Assert.Equal() Failure
                                 ↓ (pos 101)
Expected: ··· <Node Id="(@1 Type=A)" Category="CodeSchema_Class" CodeSchem···
Actual:   ··· <Node Id="(@1 Type=B)" Category="CodeSchema_Class" CodeSchem···
                                 ↑ (pos 101)
```

And my current theory is that the resultant XML is 'correct', just out-of-order, and if that's the case then the end of this XML fragment would have a reversed ordering of the test assemblies and that's ultimately what I'm trying to see.

Comment (jasonmalinowski) [inline src/VisualStudio/Core/Test/Progression/ProgressionTestHelpers.vb]: Is there an xUnit option to show the full text?

Comment (brettfo) [inline src/VisualStudio/Core/Test/Progression/ProgressionTestHelpers.vb]: There's no overload that allows this.  I'm not sure if there's a global variable to control this, but even if there is I'd rather not pollute other unit tests failures with it.

Comment (jasonmalinowski) [inline src/VisualStudio/Core/Test/Progression/ProgressionTestHelpers.vb]: I think we'd always want the behavior....

Comment (brettfo) [inline src/VisualStudio/Core/Test/Progression/ProgressionTestHelpers.vb]: I disagree.  This Xunit method is regularly used with large strings (think source before and after a transformation, or testing localized strings) and seeing the huge output makes parsing the log really difficult; it's usually sufficient to see just the failing location, but this is an instance where to properly debug a flaky test, we want to see everything.

**Comment by brettfo** [comment]
Any more comments/feedback, specifically from @jasonmalinowski?  I'd like to get this merged to gain the benefit of running on every PR.

**Comment by jasonmalinowski** [comment]
:+1:
